src/nxdoc/web-ui/web-ui-tutorials/how-to-use-rich-text-editor-rte-for-article-publishing.md

```diff
@@ -0,0 +1,189 @@
+
+In this page we will give you an overview of how you can use the rich text editor and configure it to your liking. You will also find some ideas to configure or extend it further.
+
+{{#> callout type='info'}}
+This how-to requires Nuxeo LTS 2021 or newer.
+{{/callout}}
+
+## Using the Rich Text Editor
+
+To test the default behavior of the rich text editor
+
+1. In a Workspace or a Folder, create one or several `Picture` documents.
+1. In a Workspace or a Folder, create a new `Note` document and keep the `HTML` format.
+1. Type your content and save it directly while being in the document view. Use the rich text editor toolbar to leverage the various options.
+
+{{!--     ### nx_asset ###
+    path: /default-domain/workspaces/Product Management/Documentation/Documentation Screenshots/NXDOC/Master/how-to-use-rte-article-publishing/rte-view.png
+    name: rte-view.png
+    1.1.3#screenshot#up_to_date
+--}}
+![rte-view.png](nx_asset://f2526b56-ac17-48d3-86fe-a4029a9520b6 ?border=true)
+
+To reference one or several pictures stored in Nuxeo
+
+1. Click on the `insert images from existing documents` [//TODO insert icon] toolbar icon.
+1. Search for your content using the quick filter (fulltext search) option, select as many images as you want and press the `Select` button.
+
+{{!--     ### nx_asset ###
+    path: /default-domain/workspaces/Product Management/Documentation/Documentation Screenshots/NXDOC/Master/how-to-use-rte-article-publishing/rte-image-referencing-search.png
+    name: rte-image-referencing-search.png
+    1.1.3#screenshot#up_to_date
+--}}
+![rte-image-referencing-search.png](nx_asset://10e93a3b-87e2-400b-8a9b-7b3ac85697a7 ?border=true)
+
+The search form, the query used to filter pictures and the results presentation are all configurable using Nuxeo Studio.
+
+## Accessing the Content of the Rich Text Editor
+
+As usual with Nuxeo, anything in the UI can be accessed through an external system by using the [REST API]({{page page='rest-api'}}).
+
+When retrieving a Note document, the content will be referenced in the corresponding document property, e.g.:
+
+```
+[...]
+"properties": {
+  "note:note": "Some text and now an image <img src=\&quot;https://nuxeo-server-url.nuxeo.com/nuxeo/nxfile/default/793dc397-2ecc-46ee-8840-92d1d88faed6/file:content/Beach.jpg\&quot;>"
+[...]
+```
+
+Any image you referenced is stored in a way that lets you parse the content and adapt it to your needs. Your own tool can easily adapt that URL to call further Nuxeo services, for example to request a particular rendition.
+
+## Configuring the Query
+
+By default, inserting images in the rich text editor will reference the current content stored in any document having the `Picture` facet.
+
+Depending on your needs, you may want to change the query to retrieve specific document types, documents in a specific state (e.g. only return approved images). You may even want users to pick only the latest version available for greater control.
```

Comment:
I think we should explain at this point that this relies on a page provider, so the reader doesn't have to wait until the instructions to figure it out. Suggestion:
```suggestion
By default, inserting images in the rich text editor will reference the current content stored in any document having the `Picture` facet. This is achieved by calling the `document_picker` page provider, which you may want to change to retrieve specific document types, documents in a specific state (e.g. only return approved images). You may even want users to pick only the latest version available for greater control.
```